.github/workflows/generate_pages.yml

```diff
@@ -0,0 +1,46 @@
+name: Deploy Yard to GitHub Pages
+
+on:
+  push:
+    branches:
+      - 'develop'  # Trigger the workflow on push to the 'develop' branch
+      - 'master'
+
+permissions:
+  pages: write      # Allow writing to the GitHub Pages
+  id-token: write   # Allow OIDC token to be issued
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 3.3.4  # Set Ruby version to 3.0.6
+
+      - name: Install dependencies
+        run: |
+          gem install bundler
+          bundle install
+
+      - name: Generate Yard documentation
+        run: |
+          bundle exec yard doc
+
+      - name: Upload artifact to GitHub Pages
+        uses: actions/upload-pages-artifact@v1
```

Comment:
Some of these actions have newer versions, might be worth setting them to latest.